infrastructure/tooling/src/importer/importer.js

```diff
@@ -1,49 +1,149 @@
+const { Database } = require('taskcluster-lib-postgres');
 const buffer = require('buffered-async-iterator');
+const { Operation } = require('./operations');
 const { Table, Blob } = require('fast-azure-storage');
-const prettyMilliseconds = require('pretty-ms');
 const glob = require('glob');
 const {postgresTableName} = require('taskcluster-lib-entities');
 const {REPO_ROOT, readRepoYAML} = require('../utils');
-const { readAzureTableInChunks, writeToPostgres, ALLOWED_TABLES, LARGE_TABLES } = require('./util');
+const { readAzureTableInChunks, writeToPostgres, ALLOWED_TABLES, LARGE_TABLES, TASKID_RANGES } = require('./util');
 
-const importer = async options => {
-  const { credentials, db } = options;
-  const tasks = [];
+const createOperations = async ({operations, config, monitor}) => {
+  const credentials = {
+    azure: {
+      accountId: config.AZURE_ACCOUNT,
+      accessKey: config.AZURE_ACCOUNT_KEY,
+    },
+  };
+
+  const db = new Database({
+    urlsByMode: {admin: config.ADMIN_DB_URL},
+    statementTimeout: false,
+    poolSize: config.CONCURRENCY,
+  });
 
-  let tables = [];
+  await makeTableOperations({operations, config, monitor, credentials, db});
+  await makeContainerOperations({operations, config, monitor, credentials, db});
+};
+
+const makeTableOperations = async ({operations, config, monitor, credentials, db}) => {
+  const tableNames = [];
   for (let path of glob.sync('services/*/azure.yml', {cwd: REPO_ROOT})) {
     const azureYml = await readRepoYAML(path);
     for (let t of azureYml.tables || []) {
-      if (ALLOWED_TABLES.includes(t)) {
-        tables.push(t);
+      tableNames.push(t);
+    }
+  }
+
+  // TODO: apply ALLOWED_TABLES
+
+  // support truncating each table exactly once
+  const truncated = {};
+  const truncateTable = tableName => {
+    if (!truncated[tableName]) {
+      truncated[tableName] = (async () => {
+        const pgTable = postgresTableName(tableName);
+        await db._withClient('admin', async client => {
+          await client.query(`truncate ${pgTable}`);
+        });
```

Comment:
Should this be wrapped a in a try catch statement to account for the case where we are not able to connect the db?